test/rust_src/src/eval-tests.el

```diff
@@ -0,0 +1,67 @@
+;;; eval-tests.el --- unit tests for src/eval.c      -*- lexical-binding: t; -*-
+
+;;; Commentary:
+
+;; Unit tests for src/eval.c.
+
+;;; Code:
+
+(require 'ert)
+
+(ert-deftest eval-tests--or-base ()
+  "Check (or) base cases"
+  (should (eq (or) nil))
+  (should (eq (or t) t))
+  (should (eq (or nil) nil))
+  (should (eq (or nil t) t))
+  (should (eq (or t nil) t)))
```

Comment:
If we want to be thorough here, we could test non-booleans and evaluation order.

```
(or 1 2) => 1
(or t (error "oh no")) => t
```

That said, this is already much improved test coverage and I think the elisp tests would catch issues too.

Reply:
Good idea.